CI: Workaround sporadic segfaults with retries

okurz opened the pull request:
Related progress issue: https://progress.opensuse.org/issues/176148

okurz committed:
CI: Workaround sporadic segfaults with retries

Related progress issue: https://progress.opensuse.org/issues/176148

Wabri reviewed (changes requested):
Works locally for me, maybe the problem is with macos itself

okurz commented:
I don't think I can fix [codecov/changes](https://github.com/openSUSE/Mojo-IOLoop-ReadWriteProcess/pull/79/checks?check_run_id=40033610265) failing. The overall project coverage did not change and I assume after this PR merged everything will be fine as well. It is just that I changed rules and when to run with what details in CI. Please review and approve or let me know what to change

perlpunk commented on .github/workflows/ci-tests.yaml:
```diff
@@ -34,17 +34,21 @@ jobs:
-          TEST_SHARED=1 TEST_SUBREAPER=1 PERL5OPT="-MDevel::Cover=-coverage,statement" prove -l t
+          # t/01_run.t can segfault, retrying multiple times
+          # See https://progress.opensuse.org/issues/180026
+          curl https://raw.githubusercontent.com/okurz/retry/refs/heads/main/retry > retry
```
```suggestion
          curl -O https://raw.githubusercontent.com/okurz/retry/refs/heads/main/retry
```

okurz commented:
I guess now CPAN is broken: https://github.com/openSUSE/Mojo-IOLoop-ReadWriteProcess/actions/runs/14308498488/job/40098020240?pr=79#step:5:831 says "Error: Can't locate Module/Build.pm in @INC"

perlpunk commented:
For some reason it doesn't install Module::Build, although it's in the cpanfile...

perlpunk commented:
There is something weird with https://github.com/shogo82148/actions-setup-perl I believe.
```
% cpanm --version
cpanm (App::cpanminus) version 1.9022 (/home/runner/work/_actions/shogo82148/actions-setup-perl/v1/bin/cpanm)
```
while the latest version on https://metacpan.org/dist/App-cpanminus is 1.7048

For my personal projects I usually use `perl:$version` images:
https://github.com/perlpunk/YAML-PP-p5/blob/master/.github/workflows/linux.yaml#L37

I was just told on IRC that it's https://metacpan.org/pod/Menlo::CLI::Compat behind the scenes: https://github.com/shogo82148/actions-setup-perl/blob/main/bin/cpanm#L11-L13

Still, it's somehow broken.

perlpunk commented:
In any case there is a workaround if we cannot get it working, just remove the step `perl Build.pL; ./Build` as we don't need that anyway.

okurz committed:
CI: Remove obsolete build step

Not needed to run tests and helps to avoid problems as observed in
https://github.com/openSUSE/Mojo-IOLoop-ReadWriteProcess/pull/79#issuecomment-2784800935

okurz commented:
> In any case there is a workaround if we cannot get it working, just remove the step `perl Build.pL; ./Build` as we don't need that anyway.

done that
